addons/knobs/src/components/types/Object.js

```diff
@@ -1,80 +1,65 @@
 import PropTypes from 'prop-types';
 import React from 'react';
-import Textarea from 'react-textarea-autosize';
-import deepEqual from 'deep-equal';
-
-const styles = {
-  display: 'table-cell',
-  boxSizing: 'border-box',
-  verticalAlign: 'middle',
-  width: '100%',
-  outline: 'none',
-  border: '1px solid #f7f4f4',
-  borderRadius: 2,
-  fontSize: 11,
-  padding: '5px',
-  color: '#555',
-  fontFamily: 'monospace',
-};
+import AceEditor from 'react-ace';
 
-class ObjectType extends React.Component {
-  constructor(...args) {
-    super(...args);
-    this.state = {};
-  }
+import 'brace';
+import 'brace/mode/javascript';
+import 'brace/theme/github';
 
-  getJSONString() {
-    const { json, jsonString } = this.state;
-    const { knob } = this.props;
+const getSpaces = level => Array.from({ length: level * 2 }, () => ' ').join('');
+
+function toString(obj, spaceLevel = 1) {
```

Comment:
Hi @danielduan . I didn't use json.stringify because I tried to add support for functions. In future we can add support for other JS features(regexp, Map, Set, and etc)